lib/drivers/insulet/insuletDriver.js

```diff
@@ -1066,16 +1068,23 @@ module.exports = function (config) {
               })
               .with_bgInput(bg)
               .with_carbInput(carb)
-              .with_insulinOnBoard(common.fixFloatingPoint(wiz.detail.corr_units_iob + wiz.detail.meal_units_iob, 2))
               .with_insulinCarbRatio(wiz.detail.ic_ratio_used)
               .with_insulinSensitivity(wiz.detail.correction_factor_used)
-              .with_bgTarget({target: wiz.detail.target_bg,
-                               high: wiz.detail.bg_correction_threshold })
               .with_bolus(postbolus)
               .with_payload(payload)
               .with_deviceTime(wiz.deviceTime)
               .with_units(BG_UNITS)
               .set('index', wiz.log_index);
+
+            if(wiz.detail.target_bg && wiz.detail.bg_correction_threshold) {
+              postwiz = postwiz.with_bgTarget({target: wiz.detail.target_bg,
+                               high: wiz.detail.bg_correction_threshold });
+            }
+
+            if(wiz.detail.corr_units_iob && wiz.detail.meal_units_iob) {
+              postwiz = postwiz.with_insulinOnBoard(common.fixFloatingPoint(wiz.detail.corr_units_iob + wiz.detail.meal_units_iob, 2));
+            }
+
```

Comment:
No, they are actually required because the record being dropped in the .ibf we have is part of a wizard entry. My approach is to make a "best effort" at uploading what we have here. Do you think we should drop the entire record instead?